docs/docs/contributing.md

```diff
@@ -53,35 +53,73 @@ number of relatively approachable issues with the label
+it.
+
+If a change isn't discussed before the code is submitted, there is a higher chance there will be some unconsidered
+side-effect of the change, or that the change is not in alignment with the long-term vision of Marathon.
+
+Sometimes, we are overwhelmed by the never ending sea of JIRA. If you submit an issue, and don't feel it is getting the
+level of attention that it should, please find us in the
+[#marathon channel](https://mesos.slack.com/messages/C1L7D22KY/) on the Mesos community Slack.
+
+## Backporting Policy
+
+All development (new features and bug fixes) are made, first, to the master branch.
+
+If code is heavily refactored but the behavior has not been changed, these patches are generally not backported unless
+it will help reduce merge conflicts in the future.
+
+Bugs are backported to the current supported versions of Marathon. Historically, this usually tends to be the
+current stable release, plus the past 2 major releases. (IE, if `1.5.6` is the last stable release, then `1.4.x` and `1.3.x` would also
+receive the fix).
+
+New features are rarely backported; when they are, they are backported to the latest stable version of Marathon. Major
+new features are definitely not backported.
+
 ## Submitting Code Changes to Marathon
 
-- A GitHub pull request is the preferred way of submitting patch sets. Please
-  rebase your pull requests on top of the current master using
-  `git fetch origin && git rebase origin/master`, and squash your changes to a single commit as
-  described [here](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html).
-  Yes, we want you to rewrite history - the branch on which you are
-  implementing your changes is only meant for this pull request. You can
-  either rebase before or after you squash your commits, depending on how
-  you'd like to resolve potential merge conflicts. The idea behind is that we
-  don't want an arbitrary number of commits for one pull request, but exactly
-  one commit. This commit should be easy to merge onto master, therefore we
-  ask you to rebase to master.
-  
-- Please start your commit message with "Fixes #1234 - " where #1234 is the github issue number
-  that your pull request relates to. Github will automatically link this PR to the issue and make it more
-  visible to others.
+1. A GitHub pull request is the preferred way of submitting code changes.
+
+2. If a JIRA issue doesn't already exist, please open one for the proposed change in behavior (see the section above,
+   "JIRA Issues").
+
+3. Open up a discussion about the JIRA ticket (ideally, in the comments section, but, also in the
+   [#marathon channel](https://mesos.slack.com/messages/C1L7D22KY/). Seek to get buy-off and consensus on the direction
+   before making the change.
+
+4. Please rebase your pull requests on top of the current master, as needed, to integrate upstream changes and resolve
```

Comment:
This is what we had before.

Rebasing does tend produce cleaner patch sets.

Reply:
> Rebasing does tend produce cleaner patch sets.
How? We do a squash merge into master and the release branches but how I update my working branch should not matter. In fact I'm against rebasing there since it blocks others from doing nit fixes as I've done on your leader election refactoring branch.